bin/csv2xlsx.py

```diff
@@ -5,11 +5,20 @@
 """
 
 import csv
+import codecs
 import sys
 import openpyxl
 
+def peekBOM(f):
+    first3bytes = f.read(3)
+    if first3bytes == codecs.BOM_UTF8:
+        f.seek(3)
+    else:
+        f.seek(0)
+    return f
+
 def main():
-    reader = csv.reader(sys.stdin)
+    reader = csv.reader(peekBOM(sys.stdin))
```

Comment:
instead of detecting the bom you could read one row and use `row[0].lstrip(codecs.BOM_UTF8)`